read-fonts/src/tables/morx.rs

```diff
@@ -0,0 +1,340 @@
+//! The [morx (Extended Glyph Metamorphosis)](https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6morx.html) table.
+
+use super::aat::{ExtendedStateTable, LookupU16};
+
+include!("../../generated/generated_morx.rs");
+
+impl VarSize for Chain<'_> {
+    type Size = u32;
+
+    fn read_len_at(data: FontData, pos: usize) -> Option<usize> {
+        // Size in a chain is second field beyond 4 byte `defaultFlags`
+        data.read_at::<u32>(pos.checked_add(u32::RAW_BYTE_LEN)?)
+            .ok()
+            .map(|size| size as usize)
+    }
+}
+
+impl VarSize for Subtable<'_> {
+    type Size = u32;
+
+    fn read_len_at(data: FontData, pos: usize) -> Option<usize> {
+        data.read_at::<u32>(pos).ok().map(|size| size as usize)
```

Comment:
can you add a comment explaining why the default impl is inadequate?

Reply:
Added a comment noting that the default impl assumes that the length field itself is not included in the computed length. As an aside, this feels like a bit of a footgun-- I wasn't aware that the default impl existed and it would have provided buggy behavior if I tried to use it.